src/hotspot/share/classfile/javaClasses.cpp

```diff
@@ -825,59 +825,91 @@ bool java_lang_Class::_offsets_computed = false;
 GrowableArray<Klass*>* java_lang_Class::_fixup_mirror_list = NULL;
 GrowableArray<Klass*>* java_lang_Class::_fixup_module_field_list = NULL;
 
+#ifdef ASSERT
+inline static void assert_valid_fd(fieldDescriptor* fd) {
+  assert(fd->has_initial_value(), "caller should have checked this");
```

Comment:
I think it's better to renamed to `assert_valid_static_string_field` to be more specific.

Reply:
Yes - agreed.